tasks/Task.cpp

```diff
@@ -434,40 +433,45 @@ bool Task::startHook()
     if (!_remote_peer_id.get().empty())
     {
         mRemotePeerID = _remote_peer_id.get();
-        mRemotePeerAnswerReceived = false;
         future<void> wait_remote_peer_future = mWaitRemotePeerPromise.get_future();
         // Try to get contact with the remote peer and create datachannel
-        ping();
-        future_status status =
-            wait_remote_peer_future.wait_until(chrono::system_clock::now() +
-                                               chrono::seconds(static_cast<int>(_wait_remote_peer_time_out.get().toSeconds())));
-        if (status == future_status::ready)
-        {
-            wait_remote_peer_future.get();
-        }
-        else
+        while (true)
         {
-            throw runtime_error("Timeout to get contact with the remote peer");
+            ping();
+            if (wait_remote_peer_future.wait_for(
+                    chrono::microseconds(_wait_remote_peer_time_out.get().toMicroseconds())) == future_status::ready)
+            {
+                wait_remote_peer_future.get();
+                break;
+            }
+            else if (wait_remote_peer_future.wait_for(
+                    chrono::microseconds(_wait_remote_peer_time_out.get().toMicroseconds())) == future_status::timeout)
```

Comment:
This does not solve the issue of what happens if we start the two components at the same time. When this happens, it might be that the first ping is lost. This is why in my example code I had a timeout for the wait_for (which should be the ping() period) and a timeout to break out of the loop (which is the actual _wait_remote_peer_time_out)


Moreover, you call wait_for twice, where you should save the return value and test the different possible values.
This is always going wait twice. Plus, the goal of the while loop is to send the ping more than once.

Reply:
Now I got it. 
It's confuse...thanks!